File: backend/WebApi/Lunchorder.Domain/Entities/DocumentDb/LastOrder.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;

namespace Lunchorder.Domain.Entities.DocumentDb
{
    /// <summary>
    /// Represents the last order for a user
    /// </summary>
    public class LastOrder
    {
        private IEnumerable<LastOrderEntry> _lastOrderEntries;

        [JsonProperty("id")]
        public Guid Id { get; set; }

        /// <summary>
        /// Reference to the actual user order
        /// </summary>
        public Guid UserOrderHistoryId { get; set; }

        /// <summary>
        /// The time when the order was submitted
        /// </summary>
        public DateTime OrderTime { get; set; }

        /// <summary>
        /// The items associated with the order
        /// </summary>
        public IEnumerable<LastOrderEntry> LastOrderEntries
        {
            get { return _lastOrderEntries ?? (_lastOrderEntries = new List<LastOrderEntry>()); }
            set { _lastOrderEntries = value; }
        }

        /// <summary>
        /// The total price of all entries of this order, includes rules
        /// </summary>
        public double FinalPrice { get; set; }
    }
}

File: frontend/src/app/domain/dto/lastOrder.ts
import { LastOrderEntry } from './lastOrderEntry'

export class LastOrder implements app.domain.dto.ILastOrder, Serializable<LastOrder> {
        id: string;
        userOrderHistoryId: string;
        orderTime: Date;
        finalPrice: number;
        lastOrderEntries: LastOrderEntry[];

        deserialize(input: any) : LastOrder {
                this.id = input.id;
                this.userOrderHistoryId = input.userOrderHistoryId;
                this.orderTime = input.orderTime;
                this.finalPrice = input.finalPrice;
                this.lastOrderEntries = new Array<LastOrderEntry>();
                if (input.lastOrderEntries) {
                        for (var entry of input.lastOrderEntries) {
                                this.lastOrderEntries.push(new LastOrderEntry().deserialize(entry));
                        }
                }
                return this;
        }
}